### PR TITLE
DCOS-51370: NodesTable: avoid re-renders

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

Open a nodes agents page with react dev tools and [flash on rerender](https://blog.logrocket.com/make-react-fast-again-part-3-highlighting-component-updates-6119e45e6833) enabled and see that there are less then before this change
## Screenshots

| Before                                                                                                                                        | After                                                                                                                                         | After - no render if no data change                                                                                                           |
|-----------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
| Bildschirmfoto 2019-04-08 um 15 27 05 | Bildschirmfoto 2019-04-08 um 15 25 40 |                    |

### CODE REVIEW

**Comment by mperrotti** [comment]
I can't get the smoke test `"displays one row on the table [00t]"` to fail locally 🤔

File: package.json [hunk 1]
```diff
@@ -50,6 +50,8 @@
     "d3": "3.5.16",
     "date-fns": "1.30.1",
     "dygraphs": "2.1.0",
+    "fast-deep-equal": "2.0.1",
```
Comment (brandonc) [inline package.json]: We already bundle lodash.isequal... Should we just reuse that or maybe migrate other callers?

Comment (nLight) [reply package.json]: @DanielMSchmidt you might want to follow up on this 👼

Comment (DanielMSchmidt) [reply package.json]: Followup: https://github.com/dcos/dcos-ui/pull/3795

**Pull request merged**